rfd/0086/config.md

```diff
@@ -0,0 +1,180 @@
+## Configuration syntax improvements
+
+JSON as a configuration language leaves much to be desired. It has no comments, is unforgiving in editing (we've specifically written code in ContainerPilot to point out extraneous trailing commas in the config!). Any configuration language change should also support those users we know who are generating ContainerPilot configurations automatically.
+
+We will abandon JSON in favor of the more human-friendly YAML configuration language. It has a particular advantage for those users who are generating configuration because of the ubiquity of YAML-generating libraries (this is its major advantage over Hashicorp HCL). During ContainerPilot configuration loading, we can check for files in `/etc/containerpilot.d/` (a configurable location via the `-config` flag) and merge them together.
+
+The merging process is as follows:
+
+- Lexigraphically sort all the config files.
+- Multiple `service`, `health`, `sensor` blocks are unioned.
+- Keys with the same name replace those that occurred previously.
+
+For example consider the two YAML files below.
+
+1.yml
+```
+consul:
+  host: localhost:8500
+
+services:
```

Comment:
Yeah, the context was with RFD36. I was thinking about this relative to the hierarchy of "stuff" that is involved in a given project and avoiding potentially confusing name collisions.

- Project (collection of services and their metadata)
  - Service (collection of containers)
    - Instance (the container)
      - Task (process within a container)

But on second thought all tasks are being advertised as address/ports for a Service, not as individuated tasks. So maybe leaving this as `services` is better after all; it also lines up more nicely with the terminology used by Consul.